guides/v3.17.0/in-depth-topics/patterns-for-components.md

```diff
@@ -177,3 +176,13 @@ component is destroyed and the new component is created and brought in.
 Picking different components to render in response to the data allows you to
 have different template and behavior for each case. The `{{component}}` helper
 is a powerful tool for improving code modularity.
+
+---
```

Comment:
Rather than using a line divider, can we make an `h2`-level section so that people can link to it?

```suggestion
## Learn More
```